### PR TITLE
Improved headers

### PR DESCRIPTION
**Pull request by daschuer**
This PR removes all unused include directives and replaces some definition by a forward declaration. 
This task was supported by "include-what-you-use" 
Unfortunately the changes where required to do manually, because iwyu proposes to use internal headers, instead of the documented user headers. 

This has also a nice build time benefit (single thread): 
* 51m28,476s  now
* 56m14,205s  before
* 40m43,889s  before with precompiled headers.

### CODE REVIEW

File: src/controllers/controller.cpp [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
-#include <QRegularExpression>
+#include <QJSEngine>
 #include <algorithm>
 
-#include "controllers/defs_controllers.h"
+#include "controllers/scripting/legacy/controllerscriptenginelegacy.h"
 #include "moc_controller.cpp"
 #include "util/screensaver.h"
 
+
```
Comment (Holzhaus) [inline src/controllers/controller.cpp:10]: Superfluous blank line

**Comment by daschuer** [comment]
Done

**Comment by JoergAtGithub** [comment]
LGTM! I can confirm a slight build time improvement on my system. Thank you!

**Comment by ronso0** [comment]
@daschuer (or anyone else who feels confident) could you please take care of merging into main?
Thanks!

**Comment by daschuer** [comment]
merge to main is pushed.

**Comment by uklotzde** [comment]
Build fails on Qt 6.6.0

```
/home/uk/volumes/Build/cpp/mixxx/src/effects/backends/effectsbackend.cpp: In static member function ‘static QString EffectsBackend::translatedBackendName(EffectBackendType)’:
/home/uk/volumes/Build/cpp/mixxx/src/effects/backends/effectsbackend.cpp:36:25: error: incomplete type ‘QObject’ used in nested name specifier
   36 |         return QObject::tr(backendTypeNameBuiltIn); // clazy:exclude=tr-non-literal
      |                         ^~
/home/uk/volumes/Build/cpp/mixxx/src/effects/backends/effectsbackend.cpp:42:25: error: incomplete type ‘QObject’ used in nested name specifier
   42 |         return QObject::tr(backendTypeNameUnknown); // clazy:exclude=tr-non-literal
      |                         ^~
[140/830] Building CXX object CMakeFiles/mixxx-lib.dir/src/effects/backends/effectsbackendmanager.cpp.o
```

**Comment by uklotzde** [comment]
There are more missing includes.

**Comment by daschuer** [comment]
I am not able to fix it, because I have no Qt 6.6. Can you issue a pull request?

**Pull request merged**